src/communication/stream_transfer_sender.cpp

```diff
@@ -122,6 +123,8 @@ namespace cubstream
     m_sender_daemon = cubthread::get_manager ()->create_daemon_without_entry (daemon_period,
 		      new transfer_sender_task (*this),
 		      "stream_transfer_sender");
+
+    m_p_stream_ack = cubtx::master_group_complete_manager::get_instance ();
```

Comment:
This is too specific : it assumes that the complete_manager is always the 'master_group_complete_manager', but that may not always be true.
(this is also related with my comment bellow regarding stream_ack class).

We may want to have something like:
complete_manager* complete_manager::get_instance ()
and a virtual method :
virtual complete_manager* complete_manager::create ()  
which creates specialized 'manager' implemented in each flavor of 'complete_manager'
It may not clear now the complete design, so just marking with a TODO should be enough.